backend/src/akvo/lumen/admin/add_tenant.clj

```diff
@@ -161,15 +201,46 @@
 
 (defn user-representation
   [request-headers api-root email]
-  (if-let [{:strs [id] :as user} (keycloak/fetch-user-by-email request-headers
-                                                               api-root email)]
+  (if-let [user (keycloak/fetch-user-by-email request-headers api-root email)]
     {:email email
-     :user-id id}
+     :user-id (get user "id")}
     (create-new-user request-headers api-root email)))
 
-(defn setup-keycloak
-  [label email]
-  (let [{api-root :api-root :as kc} (util/create-keycloak)
+
+(defn fetch-client
+  [request-headers api-root client-id]
+  (-> (client/get (format "%s/clients" api-root)
+                  {:query-params {"clientId" client-id}
+                   :headers request-headers})
+      :body json/decode first))
+
+(defn update-client
+  [request-headers api-root {:strs [id] :as client}]
+  (client/put (format "%s/clients/%s" api-root id)
+                {:body (json/encode client)
+                 :headers request-headers}))
+
+(defn add-tenant-urls-to-client
+  [client url]
+  (-> client
+      (update-in ["webOrigins"] (fn [webOrigins]
+                                  (conj webOrigins url)))
+      (update-in ["redirectUris"] (fn [redirectUris]
+                                    (conj redirectUris (format "%s/*" url))))))
```

Comment:
I think this could be rewritten as
```clojure
(-> client
    (update "webOrigins" conj url)
    (update "redirectUris" conj (format "%s/*" url)))
```